power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/newton_raphson_se_solver.hpp

```diff
@@ -163,38 +178,506 @@ template <bool sym> class NewtonRaphsonSESolver {
+                                    auto& left_block, const auto& right_block, const auto& measured_power,
+                                    const auto& f_x_complex) {
+        auto const& block_F_T_k_w = transpose_multiply_weight(left_block, measured_power);
+
+        multiply_add_jacobian_blocks_lhs(diag_block, block_F_T_k_w, left_block);
+        multiply_add_jacobian_blocks_lhs(block, block_F_T_k_w, right_block);
+        multiply_add_jacobian_blocks_rhs(rhs_block, block_F_T_k_w, measured_power, f_x_complex);
+    }
+
+    /// @brief Fill Q^T(j,i) of LHS(i, j) from the Q(j, i) of LHS(j, i).
+    ///
+    /// @param y_bus
+    void fill_qt(YBus<sym> const& y_bus) {
+        iterate_matrix_skip_fills(
+            [this](Idx /* row */, Idx /* col */, Idx data_idx, Idx data_idx_transpose) {
+                auto& block = data_gain_[data_idx];
+
+                block.qt_P_theta() = data_gain_[data_idx_transpose].q_P_theta();
+                block.qt_P_v() = data_gain_[data_idx_transpose].q_Q_theta();
+                block.qt_Q_theta() = data_gain_[data_idx_transpose].q_P_v();
+                block.qt_Q_v() = data_gain_[data_idx_transpose].q_Q_v();
+            },
+            y_bus);
+    }
+
+    /// @brief Process Lagrange Multiplier eta_i(i) = sum( Q^T(j,i) * phi(j) ) for j = 1 to n_bus
+    ///
+    /// @param y_bus
+    void process_lagrange_multiplier(YBus<sym> const& y_bus) {
+        iterate_matrix_skip_fills(
+            [this](Idx row, Idx col, Idx data_idx, Idx /* data_idx_transpose */) {
+                auto& block = data_gain_[data_idx];
+                auto& rhs_block = delta_x_rhs_[row];
+
+                rhs_block.eta_theta() +=
+                    dot(block.qt_P_theta(), x_[col].phi_p()) + dot(block.qt_P_v(), x_[col].phi_q());
+                rhs_block.eta_v() += dot(block.qt_Q_theta(), x_[col].phi_p()) + dot(block.qt_Q_v(), x_[col].phi_q());
+            },
+            y_bus);
+    }
+
+    template <typename Func>
+        requires std::invocable<Func, Idx /*row*/, Idx /*col*/, Idx /*data_idx*/, Idx /*data_idx_transpose*/>
+    void iterate_matrix_skip_fills(Func func, YBus<sym> const& y_bus) {
+        auto const& row_indptr = y_bus.row_indptr_lu();
+        // loop data index, all rows and columns
+        for (Idx row = 0; row != n_bus_; ++row) {
+            for (Idx data_idx = row_indptr[row]; data_idx != row_indptr[row + 1]; ++data_idx) {
+                // skip for fill-in
+                if (y_bus.map_lu_y_bus()[data_idx] == -1) {
+                    continue;
+                }
+
+                Idx const col = y_bus.col_indices_lu()[data_idx];
+                Idx const data_idx_transpose = y_bus.lu_transpose_entry()[data_idx];
+
+                func(row, col, data_idx, data_idx_transpose);
+            }
+        }
+    }
+
+    /// @brief G(row,row) += w_k
+    /// eta(row) += w_k . (z_k - f_k(x))
```

Comment:
Here and below: the dot product is denoted with `.`, which is reasonable. However, in other existing documents, e.g.,  `iterative_linear_se_solver.hpp`, the dot product is denoted simply with `*`. Not sure if either one is better (in my view it should be \cdot) but better to be consistent?

Reply:
imho out of scope of this PR. it's going to be a huge change to change one or the other and it's only for internal documentation anyways